logprep/util/log_aggregator.py

```diff
@@ -99,14 +94,3 @@ def _perform_logging_if_possible(cls):
             else:
                 if time() - cls.logs[log_id]["first_record"].created >= cls.log_period:
                     cls.logs[log_id]["aggregate"] = False
-
-    @staticmethod
```

Comment:
Logprep will crash without "filter". It is required to perform the aggregation with the logger.
The remaining methods that you removed did not seem to be called anywhere.